tests/unit/metadata/test_metadata_pkg_resources.py

```diff
@@ -71,13 +72,13 @@ def require(self, name: str) -> None:
 )
 def test_get_distribution(ws: _MockWorkingSet, req_name: str) -> None:
     """Ensure get_distribution() finds all kinds of distributions."""
-    dist = Environment(ws).get_distribution(req_name)
+    dist = Environment(cast(WorkingSet, ws)).get_distribution(req_name)
```

Comment:
The way to avoid this cast, as well as the import ordering issue, is to define `_MockWorkingSet` in at TYPE_CHECKING block:

```
if typing.TYPE_CHECKING:
    from pkg_resources import WorkingSet as _MockWorkingSet
else:
    class _MockWorkingSet(list[mock.Mock]):
        def require(self, name: str) -> None:
            pass
```